apis/v1/grpcroute_types.go

```diff
@@ -137,12 +137,14 @@ type GRPCRouteSpec struct {
 	// Support: Core
 	//
 	// +optional
+	// +listType=set
```

Comment:
Pre-existing validation that ensures that these are unique?

Reply:
Even if the change is technically breaking, I think the key questions are:
1. Does it make sense with duplicate values in this list?
2. Does the order of the values in the list matter?
3. Must the list be owned by a single SSA field manager?

If the answer to all the questions is NO, then I think it could be justified to "break" this as a bug fix.

Reply:
I agree that this could/should be considered a bug fix. To keep things clean, I'd recommend starting with `+listType=atomic` in this PR to match the current and unfortunately flawed spec today, and then file a follow up PR to fix this after this initial PR has merged. That will make it easier to point our release notes to a single small PR for the breaking change.

Reply:
Sounds like a good approach. I will update this PR.

Reply:
And here is the WIP PR that I will rebase after this PR is merged: https://github.com/kubernetes-sigs/gateway-api/pull/3972

Reply:
@robscott I have now rebased https://github.com/kubernetes-sigs/gateway-api/pull/3972. PTAL!